### PR TITLE
[Enhancement](Nereids)add RelationId to unboundRelation

### PR DESCRIPTION
**Pull request by sohardforaname**
# Proposed changes

Issue Number: close #xxx

## Problem summary

add RelationId to unboundRelation and refactor some ut used RelationUtil to allocate next RelationId.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.08 seconds
 load time: 428 seconds
 storage size: 17123343164 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221130042653_clickbench_pr_55263.html

File: fe/fe-core/src/main/java/org/apache/doris/nereids/analyzer/UnboundRelation.java [hunk 1]
```diff
@@ -42,34 +43,37 @@
  */
 public class UnboundRelation extends LogicalLeaf implements Relation, Unbound {
     private final List<String> nameParts;
+    private final RelationId id;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/analyzer/UnboundRelation.java:46]: it is better add 2 new abstract class: `AbstractLogicalRelation`, `AbstractPhysicalRelation` and put common attributes in them

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java [hunk 2]
```diff
@@ -233,9 +233,8 @@ public boolean equals(Object o) {
             return false;
         }
         GroupExpression that = (GroupExpression) o;
-        // TODO: add relation id to UnboundRelation
-        if (plan instanceof UnboundRelation) {
-            return false;
+        if (this.plan instanceof UnboundRelation) {
+            return plan.equals(that.plan);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java:237]: remove this. fix `computeLogicalProperties` in `AbstractLogicalPlan`. Let it return `UnboundLogicalProperties.INSTANCE` when current object is instance of `Unbound`

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java [hunk 3]
```diff
@@ -92,7 +93,7 @@ private LogicalPlan bindWithCurrentDb(CascadesContext cascadesContext, String ta
         Table table = getTable(dbName, tableName, cascadesContext.getConnectContext().getEnv());
         // TODO: should generate different Scan sub class according to table's type
         if (table.getType() == TableType.OLAP) {
-            return new LogicalOlapScan(cascadesContext.getStatementContext().getNextRelationId(),
+            return new LogicalOlapScan(RelationUtil.newRelationId(),
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java:96]: use unbound relation's id

**Review by morrySnow — APPROVED** [review]

**Pull request merged**